### PR TITLE
Migrate EG tests to test proxy

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/20928

### CODE REVIEW

**Review by lmazuel — APPROVED** [review]

File: sdk/eventgrid/azure-eventgrid/tests/conftest.py [hunk 1]
```diff
@@ -25,10 +25,15 @@
...
+from devtools_testutils.sanitizers import add_remove_header_sanitizer
 
 # Ignore async tests for Python < 3.5
 collect_ignore_glob = []
 if sys.version_info < (3, 5):
     collect_ignore_glob.append("*_async.py")
     collect_ignore_glob.append("test_cncf*")
+
+@pytest.fixture(scope="session", autouse=True)
+def add_aeg_sanitizer():
```
Comment (mccoyp) [inline sdk/eventgrid/azure-eventgrid/tests/conftest.py:38]: Importing `test_proxy` from `devtools_testutils` and referencing the fixture here should allow tests to use the proxy in CI:
```suggestion
def add_aeg_sanitizer(test_proxy):
```

**Pull request merged**